File: smart_farm/src/api/systemSetting.ts
import { get, http } from '@/utils/request'

/** 与 smart_plant SystemSetting 一一对应的前端配置类型。 */
export interface SystemSetting {
  id: number
  systemName: string
  logoUrl?: string
  faviconUrl?: string
  systemDescription?: string
  copyrightInfo?: string
  homeTitle: string
  loginBackgroundUrl?: string
  themeColor: string
  timezone: string
  datetimeFormat: string
  defaultPageSize: number
  maxUploadSizeMb: number
  allowedUploadTypes: string
  recordNumber?: string
  officialWebsite?: string
  contactEmail?: string
  servicePhone?: string
  version: number
  updateTime?: string
}

export type SystemSettingUpdatePayload = Omit<SystemSetting, 'id' | 'updateTime'>
export type SystemAssetType = 'logo' | 'favicon' | 'loginBackground'

interface ApiResult<T> {
  code: number
  message: string
  data: T
}

export interface SystemAssetUploadResult {
  type: SystemAssetType
  url: string
  size: number
}

/** 登录前读取公开配置，使品牌外观无需写死在前端包中。 */
export const getPublicSystemSetting = () => {
  return get<ApiResult<SystemSetting>>('/public/system-settings')
}

/** 管理员打开系统设置弹框时读取最新值和版本号。 */
export const getAdminSystemSetting = () => {
  return get<ApiResult<SystemSetting>>('/admin/system-settings')
}

/** 整体保存五个分区的系统设置。 */
export const updateSystemSetting = (data: SystemSettingUpdatePayload) => {
  return http<ApiResult<SystemSetting>>({ url: '/admin/system-settings', method: 'put', data })
}

/** 上传系统品牌图片，并由后端返回可持久化的相对地址。 */
export const uploadSystemAsset = (type: SystemAssetType, file: File) => {
  const data = new FormData()
  data.append('file', file)
  return http<ApiResult<SystemAssetUploadResult>>({
    url: '/admin/system-settings/assets',
    method: 'post',
    params: { type },
    data,
    headers: { 'Content-Type': 'multipart/form-data' },
  })
}

File: smart_plant/src/main/java/com/smart_plant/smart_plant/entity/SystemSetting.java
package com.smart_plant.smart_plant.entity;

import lombok.Data;

import java.time.LocalDateTime;

/** 后台站点唯一一条全局系统配置。 */
@Data
public class SystemSetting {

    /** 固定为1的单例主键。 */
    private Integer id;
    /** 系统名称。 */
    private String systemName;
    /** 系统Logo相对地址或HTTP(S)地址。 */
    private String logoUrl;
    /** 浏览器网站图标地址。 */
    private String faviconUrl;
    /** 系统用途描述。 */
    private String systemDescription;
    /** 页面展示的版权信息。 */
    private String copyrightInfo;
    /** 首页及浏览器标题。 */
    private String homeTitle;
    /** 登录页背景图地址。 */
    private String loginBackgroundUrl;
    /** 十六进制系统主题色。 */
    private String themeColor;
    /** IANA时区名称，例如Asia/Shanghai。 */
    private String timezone;
    /** Java日期时间格式。 */
    private String datetimeFormat;
    /** 列表分页默认条数。 */
    private Integer defaultPageSize;
    /** 文件上传大小限制，单位MB。 */
    private Integer maxUploadSizeMb;
    /** 允许上传的扩展名，统一使用逗号分隔。 */
    private String allowedUploadTypes;
    /** 网站备案号。 */
    private String recordNumber;
    /** 官网HTTP(S)地址。 */
    private String officialWebsite;
    /** 站点联系邮箱。 */
    private String contactEmail;
    /** 站点客服电话。 */
    private String servicePhone;
    /** 用于防止并发覆盖的乐观锁版本号。 */
    private Integer version;
    /** 最后修改用户ID。 */
    private Long updateBy;
    /** 配置创建时间。 */
    private LocalDateTime createTime;
    /** 配置最后修改时间。 */
    private LocalDateTime updateTime;
}
